olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -575,15 +584,28 @@ uint64_t DefaultCacheImpl::MaybeEvictData(leveldb::WriteBatch& batch) {
     const auto& key = it->key();
     const auto& properties = it->value();
 
-    evicted += key.size() + properties.size;
-    batch.Delete(key);
+    const auto expiry_key = CreateExpiryKey(key);
+    auto current_eviction_size = key.size() + properties.size;
+    if (IsExpiryValid(properties.expiry)) {
+      current_eviction_size += expiry_key.size() + kExpiryValueSize;
+    }
+
+    if (eviction_limit && evicted + current_eviction_size > eviction_limit) {
+      OLP_SDK_LOG_INFO_F(
```

Comment:
You can imagine how these logs will flood the stdout on a 1GB eviction. Put it to debug, please.
Only one final log at the end with total numbers should suffice.